### DISCUSSION TITLE
[Bug]: npm run update script crashed: docs\dev\single-compose.yml: The system cannot find the path specified

### DISCUSSION THREAD

**Discussion by myxgenius** [danny-avila/LibreChat#2768] [Troubleshooting] [answered]
### What happened?

I tried to npm run update and select 'using docker' and 'not using default docker-compose file' and it crashed.

Background: I made some local changes to the source code and want to try them out. I used to use the npm run update script with y and n selection for the prompt to build the image so that i can launch the server with docker compose up, but it stopped working at some point. I then tried a fresh pull of the project and run npm run update, and the issue persists without any of my local changes.
 
System: Windows 11 23H2

### Steps to Reproduce

1. git clone https://github.com/danny-avila/LibreChat.git
2. cd LibreChat
3. cp .env.example .env
4. npm run update
5. > y and n (select using docker, not using default docker-compose file)

### Relevant log output

```shell
PS T:\> cd .\LibreChat\
PS T:\LibreChat> cp .env.example .env
PS T:\LibreChat> npm run update

> LibreChat@0.7.2 update
> node config/update.js

Are you using Docker? (y/n):
> y
Are you using the default docker-compose file? (y/n):
> n
Starting update script, this may take a minute or two depending on your system and network.
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
Fetching the latest repo...
Switching to main branch...
Already on 'main'
Your branch is up to date with 'origin/main'.
Pulling the latest code from main...
From https://github.com/danny-avila/LibreChat
 * branch              main       -> FETCH_HEAD
Already up to date.
Removing previously made Docker container...
docker compose -f ./docs/dev/single-compose.yml down
CreateFile T:\LibreChat\docs\dev\single-compose.yml: The system cannot find the path specified.
node:internal/errors:932
  const err = new Error(message);
              ^

Error: Command failed: docker compose -f ./docs/dev/single-compose.yml down
    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at T:\LibreChat\config\update.js:86:5
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 14,
  signal: null,
  output: [ null, null, null ],
  pid: 33276,
  stdout: null,
  stderr: null
}

Node.js v20.10.0
PS T:\LibreChat>
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment, marked as answer]
Try to make a `docker-compose.override.yml` file in the root folder of the project, with this content:
```yaml
version: '3.4'

services:
# LOCAL BUILD
  api:
    image: librechat
    build:
      context: .
      target: node
```

Then run docker compose up, LibreChat should build with your local changes, using the regular docker-compose file.
You can also override other parameters using the override file, see this for more info: https://www.librechat.ai/docs/configuration/docker_override

**Reply by myxgenius** [reply]
Ah I see, Thanks for the reply!
